### PR TITLE
Fix graph evaluator panics

### PR DESCRIPTION
**Pull request by aliscott**

### CODE REVIEW

**Comment by aliscott** [comment]
Test failures look unrelated.

File: internal/hcl/block.go [hunk 1]
```diff
@@ -1132,6 +1132,10 @@ func (b *Block) FullName() string {
 }
 
 func (b *Block) Type() string {
+	if b == nil || b.HCLBlock == nil {
+		return ""
+	}
+
```
Comment (hugorut) [inline internal/hcl/block.go:1138]: hmm this means that somewhere else in the code we've introduced logic which creates a nil `Block` which is something that shouldn't happen. I'm ok with this fix as it's very hard to track down where this happens, but we should keep this in mind for any errors/price inaccuracies we see later with the graph ev.

**Review by hugorut — APPROVED** [review]
This looks good, however a small nitpick, in future could we please add some contextual information to the PR comment/description for:

1. what the problem was
2. why this fix

Otherwise it's very tricky for me to review these.

🙏

File: internal/hcl/block.go (hunk 1 above)
Comment (aliscott) [reply internal/hcl/block.go:1138]: Yep, I thought this was odd too, but we seem to have this fix in a lot of places 😕

**Pull request merged**